docs/asyncapi.yaml

```diff
@@ -0,0 +1,158 @@
+asyncapi: 2.4.0
+info:
+  title: FFC Payment Statement Data
+  version: '1.0.0'
+  description: Publish datasets from Data Warehouse for Payment Statements
+channels:
+  organisation:
+    publish:
+      message:
+        $ref: '#/components/messages/Organisation'
+  calculation:
+    publish:
+      message:
+        $ref: '#/components/messages/Calculation'
+components:
+  messages:
+    Organisation:
+      name: Organisation
+      contentType: application/json
+      payload:
+        $ref: '#/components/schemas/Organisation'
+    Calculation:
+      name: Calculation
+      contentType: application/json
+      payload:
+        $ref: '#/components/schemas/Calculation'
+  schemas:
+    Organisation:
+      type: object
+      properties:
+        name:
+          $ref: '#/components/schemas/BusinessName'
+        sbi:
+          $ref: '#/components/schemas/SBI'
+        frn:
+          $ref: '#/components/schemas/FRN'
+        addressLine1:
+          $ref: '#/components/schemas/AddressLine1'
+        addressLine2:
+          $ref: '#/components/schemas/AddressLine2'
+        addressLine3:
+          $ref: '#/components/schemas/AddressLine3'
+        city:
+          $ref: '#/components/schemas/City'
+        county:
+          $ref: '#/components/schemas/County'
+        postcode:
+          $ref: '#/components/schemas/Postcode'
+        emailAddress:
+          $ref: '#/components/schemas/Email'
+        updated:
+          $ref: '#/components/schemas/Updated'
+    Calculation:
+      type: object
+      properties:
+        calculationId:
+          $ref: '#/components/schemas/CalculationId'
```

Comment:
no `CalculationId` but there is `BusinessName` which isn't used...

Reply:
if `CalculationId` is something which DWH gives us, do we wish to use it in our DB?

i.e. if it gives us `CalculationId` 4 for our fresh new service do we save down 4 or allow Sequelize to auto-increment and assign it 1? Similar thing to how we delete `paymentRequestId` coming in from any of the Topics.

This hassa maybe business-y use of coupling and keeping DWH ids the same as our internal services?

Can we be sure that DWH will not produce a duplicate `CalculationId`? Is there a duplication case for replayed updates?

Reply:
> re is `BusinessName` which isn't

Added `CalculationId`.  BusinessName is referenced by the `name` property in the `organisation` message payload.

Reply:
> if CalculationId is something which DWH gives us, do we wish to use it in our DB?

> i.e. if it gives us CalculationId 4 for our fresh new service do we save down 4 or allow Sequelize to auto-increment and assign it 1? Similar thing to how we delete paymentRequestId coming in from any of the Topics.

We're not saving any data in this database other than updating the date the dataset was published.  DWH will be updating this database themselves direct from Informatica.

I'd expect the schema to evolve to best suit them but they've initially indicated that this initial schema should work for them.

> This hassa maybe business-y use of coupling and keeping DWH ids the same as our internal services?

The purpose of this service publishing data is so we can sanitize and convert that data into a format that best works for our consuming services whilst keeping the database what best works for DWH.  So I'm not expecting an exact 1-2-1 mapping.  This first draft is just showing what that 1-2-1 mapping would look like so we can see how close it is to what we want.  Before this PR is merged, I'd expect it refined.

> Can we be sure that DWH will not produce a duplicate CalculationId? Is there a duplication case for replayed updates?

CalculationId should be unique for every calculation in Siti Agri, just like the invoice number associated with it.  For context the calculation Id is an incrementing number and we won't get sequential values because not every calculation will result in payment.  What I don't know is whether calculation Id is unique across schemes.  If so then once we add datasets for other Siti Agri stuff then we'll need to agree a tweak with DWH for later in the year.  Either by changing the PK or by converting it to a composite key with `scheme`.

DWH will handle not adding duplicate calculations on their side.